.github/workflows/test.yml

```diff
@@ -0,0 +1,88 @@
+name: Test
+on: [push, pull_request]
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        gemfile:
+          - Gemfile
+          - gemfiles/Gemfile-rails-6-0
+          - gemfiles/Gemfile-rails-5-2
+          - gemfiles/Gemfile-rails-5-1
+          - gemfiles/Gemfile-rails-5-0
+          - gemfiles/Gemfile-rails-4-2
+          - gemfiles/Gemfile-rails-4-1
+        ruby:
+          - 2.4
+          - 2.5
+          - 2.6
+          - 2.7
+          - 3.0
+        exclude:
+          - ruby: 2.4
+            gemfile: Gemfile
+          - ruby: 2.4
+            gemfile: gemfiles/Gemfile-rails-6-0
+          - ruby: 2.4
+            gemfile: gemfiles/Gemfile-rails-4-1
+          - ruby: 2.5
+            gemfile: gemfiles/Gemfile-rails-4-1
+          - ruby: 2.6
+            gemfile: gemfiles/Gemfile-rails-4-1
+          - ruby: 2.6
+            gemfile: gemfiles/Gemfile-rails-4-2
+          - ruby: 2.7
+            gemfile: gemfiles/Gemfile-rails-4-1
+          - ruby: 2.7
+            gemfile: gemfiles/Gemfile-rails-4-2
+          - ruby: 2.7
+            gemfile: gemfiles/Gemfile-rails-5-0
+          - ruby: 2.7
+            gemfile: gemfiles/Gemfile-rails-5-1
+          - ruby: 2.7
+            gemfile: gemfiles/Gemfile-rails-5-2
+          - ruby: 3.0
+            gemfile: gemfiles/Gemfile-rails-4-1
+          - ruby: 3.0
+            gemfile: gemfiles/Gemfile-rails-4-2
+          - ruby: 3.0
+            gemfile: gemfiles/Gemfile-rails-5-0
+          - ruby: 3.0
+            gemfile: gemfiles/Gemfile-rails-5-1
+          - ruby: 3.0
+            gemfile: gemfiles/Gemfile-rails-5-2
+    runs-on: ubuntu-latest
+    env:
+      BUNDLE_GEMFILE: ${{ matrix.gemfile }}
+      UPLOADCARE_PUBLIC_KEY: demopublickey
+      UPLOADCARE_SECRET_KEY: demoprivatekey
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Bundler 1.x for Rails 4.x
+        if: ${{ matrix.gemfile == 'gemfiles/Gemfile-rails-4-1' || matrix.gemfile == 'gemfiles/Gemfile-rails-4-2' }}
+        run: echo "BUNDLER_VERSION=1.17.3" >> $GITHUB_ENV
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true # runs bundle install and caches installed gems automatically
+          bundler: ${{ env.BUNDLER_VERSION || 'latest' }}
+      - run: bundle exec rake
+
+  code-analysis:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      matrix:
+        ruby-version: ['2.7',]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby-version }}
+        bundler-cache: true
+    - name: Install gems
+      run: gem install rubocop
+    - name: Check code
+      run: bundle exec rubocop
```

Comment:
This should run first, and, ideally, fail (when we'll fix all style errors).

Reply:
Github runs jobs in parallel by default, and I think, this is OK - this reduces the overall workflow duration. But we can set the order.

Reply:
there is no point running tests if linter fails. we're paying for wasted resources :)

Reply:
It is a quite rare situation because most of Ruby developers see rubocop errors in IDE or use `overcommit`.
The parallel run makes an entire workflow faster.

But again - we can setup this as you want (I see that you already wrote a config :slightly_smiling_face: )

Reply:
even if its rare, does not mean that will not happen.
and linting is super fast anyway